src/tree/treemath.rs

```diff
@@ -96,71 +97,72 @@ pub(crate) fn sibling(index: NodeIndex, size: LeafIndex) -> Result<NodeIndex, Tr
     }
 }
 
-// Ordered from leaf to root
-// Includes neither leaf nor root
-pub(crate) fn dirpath(index: NodeIndex, size: LeafIndex) -> Result<Vec<NodeIndex>, TreeMathError> {
-    let r = root(size);
-    if index == r {
-        return Ok(vec![]);
-    }
-
-    let mut d = vec![];
-    let mut x = parent(index, size)?;
-    while x != r {
-        d.push(x);
-        x = parent(x, size)?;
-    }
-    Ok(d)
-}
-
-// Ordered from leaf to root
-// Includes leaf and root
-pub(crate) fn dirpath_long(
-    index: NodeIndex,
+/// Direct path from a leaf node to the root.
+/// Does not include the leaf node but includes the root.
+pub(crate) fn leaf_direct_path(
+    leaf_index: LeafIndex,
     size: LeafIndex,
 ) -> Result<Vec<NodeIndex>, TreeMathError> {
+    let node_index = NodeIndex::from(leaf_index);
     let r = root(size);
-    if index == r {
+    if node_index == r {
         return Ok(vec![r]);
     }
 
-    let mut x = index;
-    let mut d = vec![index];
+    let mut d = vec![];
+    let mut x = node_index;
     while x != r {
         x = parent(x, size)?;
         d.push(x);
     }
     Ok(d)
 }
 
-// Ordered from leaf to root
-// Includes root but not leaf
-pub(crate) fn direct_path_root(
-    index: NodeIndex,
+/// Direct path from a parent node to the root.
+/// Includes the parent node and the root.
+/// Returns an error if the `index` is not a parent node.
+pub(crate) fn parent_direct_path(
+    node_index: NodeIndex,
     size: LeafIndex,
 ) -> Result<Vec<NodeIndex>, TreeMathError> {
```

Comment:
While you're here, you can remove the `Result` and add an `unwrap` at the `parent` call. It's never going to throw an error, because we're never calling it with the root node index.

Reply:
It can throw an error if called with a node that's not a parent.

Reply:
True. I guess that observation is valid for the generic implementation of `direct_path` as opposed to the specific leaf- and parent variants.

Reply:
Exactly, I'd have to remove the check for whether it's really a parent node, which I'd rather not do given how the function is supposed to be used.